cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -399,6 +400,12 @@ func ValidateIPNetFromString(subnetStr string, minAddrs int64, isDualStack bool,
 		if numAddresses < minAddrs {
 			allErrs = append(allErrs, field.Invalid(fldPath, s.String(), fmt.Sprintf("subnet with %d address(es) is too small, the minimum is %d", numAddresses, minAddrs)))
 		}
+
+		// Warn when the subnet is in site-local range - i.e. contains addresses that belong to FEC0::/10
+		_, siteLocalNet, _ := net.ParseCIDR("FEC0::/10")
+		if siteLocalNet.Contains(s.IP) || s.Contains(siteLocalNet.IP) {
+			klog.Warningf("the subnet %v contains IPv6 site-local addresses that belong to FEC0::/10 which is deprecated", s)
```

Comment:
klog.Warningf->klog.InfoS, see more detail in
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging

Reply:
@lala123912
Do you mean that we should change it to 
``` 
klog.InfoS("Warning: the subnet contains IPv6 site-local addresses that belong to FEC0::/10 which is deprecated", "subnet", s)
```
?

Reply:
yeah

Reply:
we are not moving moving to structured logging yet, so please use klog.Warning for the time being.
also klog.Warning is not being removed anytime soon.

/hold

Reply:
we have a number of `migrate ....` PRs about this:
https://github.com/kubernetes/kubernetes/pulls?q=is%3Apr+is%3Aopen+label%3Aarea%2Fkubeadm

the SIG Instrumentation did not reply to my questions on this topic.

Reply:
OK, I prefer the warning here.
Updated.

Hope it is the last update 😊